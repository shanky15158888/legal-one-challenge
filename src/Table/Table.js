import React from 'react'
import { Link } from 'react-router-dom'

export default function Table({ rows, cols }) {
    return (
        <>
            <table className="table table-striped">
                <thead>
                    <tr>
                        {
                            cols && cols.length? cols.map((col, index) => {
                                return <th key={index} scope="col">{col}</th>
                            }): null
                        }
                    </tr>
                </thead>
                <tbody>
                    {
                        rows && rows.length? rows.map((row, index) => {
                            return <tr key={index} scope="col">
                                {
                                    row.data && row.data.length ? row.data.map((item, subIndex) => {
                                        if(item.type === "link") {
                                            return <td key={subIndex}><Link to={item.url}>{item.data}</Link></td>
                                        } else {
                                            return <td key={subIndex}>{item.data}</td>
                                        }
                                    }): null
                                }
                            </tr>
                        }): null
                    }
                </tbody>
            </table>
        </>
    )
}
