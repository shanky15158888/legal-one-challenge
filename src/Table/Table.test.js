import React from 'react';
import renderer from 'react-test-renderer';
import Table from './Table';
import { BrowserRouter as Router } from "react-router-dom";

test('Table component testing', () => {
    const cols = ["Phone number", "Number of calls", "Last call details"]
    const rows = [
        { data: [
            {data: "+49151484522"},
            {data: "05/10/2020 20:10:00"},
            {data: "need reschedule"}
        ]}
    ]
    const component = renderer.create(
        <Router>
            <Table cols={cols} rows={rows}/>
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});