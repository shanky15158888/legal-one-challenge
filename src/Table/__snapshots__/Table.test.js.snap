// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component testing 1`] = `
<table
  className="table table-striped"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Phone number
      </th>
      <th
        scope="col"
      >
        Number of calls
      </th>
      <th
        scope="col"
      >
        Last call details
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      scope="col"
    >
      <td>
        +49151484522
      </td>
      <td>
        05/10/2020 20:10:00
      </td>
      <td>
        need reschedule
      </td>
    </tr>
  </tbody>
</table>
`;
